run_resnet.py

```diff
@@ -0,0 +1,94 @@
+from argparse import ArgumentParser
+
+import numpy as np
+import torch
+from pytorch_lightning import Trainer
+from torch.utils.data import DataLoader
+
+from callbacks import get_callbacks
+from config import Config
+from constants import NO_VAL
+from dataloader import ChestXrayDataset, ChestXrayTestDataset
+from resnet import LitResnet
+
+# Get them from command line!
+BATCH_SIZE = 32
+PATCH_SIZE = 128
+MAX_EPOCHS = 10
+# 3e-4 for smaller batch-size
+# 1e-3 or 1e-4 for bigger batch-size
+LEARNING_RATE = 3e-4
+WEIGHT_DECAY = 1e-5
+N_PATCHES = 50
+RESNET_VERSION = 50
+OUT_FEATURES = 1
+
+
+# For AdamW
+# learning rate proportional to square root of batch_size (theoretically)
+
+
+def run_resnet():
+    parser = ArgumentParser()
+    parser = Trainer.add_argparse_args(parser)
+    args = parser.parse_args(["--gpus=0", "--max_epochs=20", "--val_check_interval=40"])
+
+    train_dataset = ChestXrayDataset(phase="train", crop=True, patch_size=PATCH_SIZE)
+    val_dataset = ChestXrayDataset(phase="val", crop=True, patch_size=PATCH_SIZE)
+    test_dataset = ChestXrayTestDataset(crop=True, patch_size=PATCH_SIZE, n_per_image=N_PATCHES)
+
+    train_dataloader = DataLoader(train_dataset, batch_size=BATCH_SIZE, shuffle=True)
+    val_dataloader = DataLoader(val_dataset, batch_size=BATCH_SIZE)
+    test_dataloader = DataLoader(test_dataset, batch_size=1)
+
+    config = Config(
+        batch_size=BATCH_SIZE,
+        patch_size=PATCH_SIZE,
+        max_epochs=MAX_EPOCHS,
+        learning_rate=LEARNING_RATE,
+        weight_decay=WEIGHT_DECAY,
+        n_patches=N_PATCHES,
+        len_train_dataset=len(train_dataloader),
+        no_val=NO_VAL,
+    )
+    trainer = Trainer.from_argparse_args(args, callbacks=get_callbacks())
+    model = LitResnet(out_features=OUT_FEATURES, config=config, resnet_version=RESNET_VERSION)
+    trainer.fit(model, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader)
+    model.save_configs(log_dir=trainer.log_dir)
+    i = 0
+    outputs = []
+    while i < N_PATCHES:
+        output = trainer.predict(
+            model,
+            test_dataloader,
+            ckpt_path="best" if not NO_VAL else None,
+        )
+        outputs.append(output)
+        i += 1
```

Comment:
I know, I know, for loops get boring. But from a readability standpoint, this really *should* be a for loop over `range(N_PATCHES)`. When an experienced developer sees a `while` loop, they except something more complicated.

Reply:
Okay